org.eclipse.scanning.api/src/org/eclipse/scanning/api/ui/auto/IModelViewer.java

```diff
@@ -40,7 +40,7 @@
 	 * @param model
 	 * @throws exception if interface cannot be rendered from this model.
 	 */
-	void setModel(T model) throws InterfaceInvalidException;
+	void setModel(T model) throws Exception;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)